### PR TITLE
Dust Apps: Prevent datasets to have the same key hack

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

This is a hack to fix this: https://github.com/dust-tt/dust/issues/11292

Since keys and data are not linked, if you name a key "foo" and you want to name a second key "foo2", when you start typing  the second name at some point you will have two "foo" and that messes up the data structure of the dataset. 

This is hacky but it's such an edge case that I think it's acceptable. 

## Tests

Locally. 

## Deploy Plan

Deploy front.

### CODE REVIEW

File: front/components/app/DatasetView.tsx [hunk 1]
```diff
@@ -643,7 +650,7 @@ export default function DatasetView({
                           "font-mono w-full resize-none border-0 bg-transparent px-1 py-0 text-[13px] font-normal italic placeholder-gray-400 ring-0 focus:ring-0",
                           readOnly
                             ? "text-gray-500"
-                            : "text-gray-700 dark:text-gray-700-night"
+                            : "dark:text-gray-700-night text-gray-700"
```
Comment (PopDaph) [inline front/components/app/DatasetView.tsx:653]: This and below are just from automatic linting, classes were not in the right order. 💅🏻

**Review by spolu — APPROVED** [review]
I'm not sure I managed to reproduce if I name a key foo and then edit a second key when I hit the foo state I indeed have 2 foos which seem to link to the same data but if I continue typing everything works as expected?

**Comment by PopDaph** [comment]
Seen IRL!
